src/Entity/Activity.php

```diff
@@ -0,0 +1,222 @@
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="App\Repository\ActivityRepository")
+ */
+class Activity
+{
+    /**
+     * @ORM\Id()
+     * @ORM\GeneratedValue()
+     * @ORM\Column(type="integer")
+     */
+    protected $id;
+
+    /**
+     * @ORM\Column(type="string", length=30)
+     */
+    private $name;
+
+    /**
+     * @ORM\Column(type="string", length=1000)
+     */
+    private $description;
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $applicationDeadline;
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $finalDeadline;
+
+    /**
+     * @ORM\Column(type="string", length=30)
+     */
+    private $status;
+
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private $idOwner;
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $createdAt;
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $updatedAt;
+
+    /**
+     * @var \Doctrine\Common\Collections\Collection|Technology[]
+     * @ORM\ManyToMany(targetEntity="Technology")
+     */
+    protected $activityTechnologies;
+
+    /**
+     * @var \Doctrine\Common\Collections\Collection|Type[]
+     * @ORM\ManyToMany(targetEntity="Type")
+     */
+    protected $activityTypes;
+
+    public function __construct()
+    {
+        $this->activityTechnologies = new ArrayCollection();
+        $this->activityTypes = new ArrayCollection();
+    }
+
+    /**
+     * @param Technology $activityTechnology
+     */
+    public function addActivityTechnology(Technology $activityTechnology)
+    {
+        if ($this->activityTechnologies->contains($activityTechnology)) {
+            return;
+        }
+        $this->activityTechnologies->add($activityTechnology);
+    }
+
+    /**
+     * @param Technology $activityTechnology
+     */
+    public function removeActivityTechnology(Technology $activityTechnology)
+    {
+        if (!$this->activityTechnologies->contains($activityTechnology)) {
+            return;
+        }
+        $this->activityTechnologies->removeElement($activityTechnology);
+    }
+
+    /**
+     * @param Type $activityType
+     */
+    public function addActivityType(Type $activityType)
+    {
+        if ($this->activityTypes->contains($activityType)) {
+            return;
+        }
+        $this->activityTypes->add($activityType);
+    }
+
+    /**
+     * @param Type $activityType
+     */
+    public function removeActivityType(Type $activityType)
+    {
+        if (!$this->activityTypes->contains($activityType)) {
+            return;
+        }
+        $this->activityTypes->removeElement($activityType);
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name): self
+    {
+        $this->name = $name;
+
+        return $this;
```

Comment:
Not all functions return `$this`. Example: `addActivityTechnology`, `removeActivityTechnology`.